bot/cogs/help.py

```diff
@@ -107,12 +106,26 @@ def _get_query(self, query: str) -> Union[Command, Cog]:
         if command:
             return command
 
-        cog = self._bot.cogs.get(query)
-        if cog:
+        # Find all cog categories that match.
+        cogs = self._bot.cogs.values()
+        cog_matches = [cog for cog in cogs if hasattr(cog, "category") and cog.category == query]
+
+        # Try to search by cog name if no categories match.
+        if not cog_matches:
+            cog = self._bot.cogs.get(query)
+
+            # Don't consider it a match if the cog has a category.
+            if cog and not hasattr(cog, "category"):
+                cog_matches = [cog]
+
+        if cog_matches:
+            cog = cog_matches[0]
+            cmds = (cog.get_commands() for cog in cog_matches)  # Commands of all cogs
+
             return Cog(
-                name=cog.qualified_name,
-                description=inspect.getdoc(cog),
-                commands=[c for c in self._bot.commands if c.cog is cog]
+                name=cog.category if hasattr(cog, "category") else cog.qualified_name,
+                description=cog.description,
```

Comment:
This uses the docstringoif the first cog in the list of matches which isn't great but it is simple. What would be a good way to define a docstring for a specific category?

Reply:
That reminds me actually, I was going to suggest also to move the category into scope of the class instead, as it's not instance specific.

One approach regarding the description is the instead find the first instance of another attribute `category_description` across the cogs, so you could define it once anywhere and the rest won't need to. 

```py
description=cog.category_description if hasattr(cog, "category_description") else cog.description
```

Reply:
Here's a completely different idea:

Keep a dictionary which maps categories to cogs (and also stores the description). This would be an attribute of `bot`, similar to how d.py already has a cogs dictionary. Thus, categories would be defined in extensions (well it could be anywhere outside the cog class really). The current solution just seems too hacky.

Reply:
@scragly Thoughts on that? Also, what was the other thing you were talking about on Discord regarding the help command when you first had a look at the diff?

Reply:
Discussion on Discord alleviated my concerns about this being "hacky", especially if the attribute is changed from an instance attribute to a class attribute. Additionally, the support for categories will be documented in the `HelpSession` class.

Regarding the description, I agree making it an attribute of the cog is a good idea.

If categories are going to be used more frequently in the code base, a more "structured" and documented approach could be considered. For now, with only one category, this is adequate.